cpp/include/cudf/strings/repeat_strings.hpp

```diff
@@ -76,15 +84,91 @@ std::unique_ptr<string_scalar> repeat_strings(
+ * maximum value that can be indexed by size_type (offset_type). In such situations, an exception
+ * may be thrown, or the output result is undefined. As such, the caller is responsible for checking
+ * output overflow to prevent runtime exception and data corruption.
+ *
+ * @code{.pseudo}
+ * Example:
+ * strs         = ['aa', null, '', 'bbc-']
+ * repeat_times = [ 1,   2,     3,  4   ]
+ * out          = repeat_strings(strs, repeat_times)
+ * out is ['aa', null, '', 'bbc-bbc-bbc-bbc-']
+ * @endcode
+ *
+ * @throw cudf::logic_error if the input `repeat_times` column has data type other than integer.
+ * @throw cudf::logic_error if the input columns have different sizes.
+ *
+ * @param input The column containing strings to repeat.
+ * @param repeat_times The column containing numbers of times that the corresponding input strings
+ *        are repeated.
+ * @param output_strings_sizes The optional column containing pre-computed sizes of the output
+ *        strings.
+ * @param mr Device memory resource used to allocate the returned strings column.
+ * @return New column containing the repeated strings.
+ */
+std::unique_ptr<column> repeat_strings(
+  strings_column_view const& input,
+  column_view const& repeat_times,
+  std::optional<column_view> output_strings_sizes = std::nullopt,
+  rmm::mr::device_memory_resource* mr             = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Compute sizes of the output strings if each string in the input strings column
+ * is repeated by the numbers of times given in another numeric column.
+ *
+ * During calculating the string output sizes, these sizes are also sum up (stored in an int64_t
+ * number) and returned, which can be used to detect if the input strings column can be safely
+ * repeated without data corruption due to overflow in string indexing.
+ *
+ * @code{.pseudo}
+ * Example:
+ * strs         = ['aa', null, '', 'bbc-']
+ * repeat_times = [ 1,   2,     3,  4   ]
+ * out          = repeat_strings_output_sizes(strs, repeat_times)
+ * out is [2, 0, 0, 16]
+ * @endcode
+ *
+ * @throw cudf::logic_error if the input `repeat_times` column has data type other than integer.
+ * @throw cudf::logic_error if the input columns have different sizes.
+ *
+ * @param input The column containing strings to repeat.
+ * @param repeat_times The column containing numbers of times that the corresponding input strings
+ *        are repeated.
+ * @param mr Device memory resource used to allocate the returned strings column.
+ * @return A pair with the first item is an int32_t column containing sizes of the output strings,
+ *         and the second item is an int64_t number containing the total sizes (in bytes) of the
+ *         output strings column.
+ */
+std::pair<std::unique_ptr<column>, int64_t> repeat_strings_output_sizes(
+  strings_column_view const& input,
+  column_view const& repeat_times,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Comment:
Seems we don't need this anymore?

Reply:
I think we still need it. If we don't have it,  we have to compute the output sizes (stored in `int32_t`) and then sum up these sizes to detect overflow. Those 2 steps are done in 2 separate passes, which involve 2 kernel launches. In addition, if there is any individual string causing overflow, the first pass truncates that overflowed value to store in an `int32_t` value thus the second pass may miss such overflow event.

I came up with an idea: use `thrust::transform_reduce` that only invokes one kernel, and it can compute output string sizes in `int64_t`  for summing up the sizes. These output sizes will be stored in an `int32_t` column anyway, but their sum is computed in `int64_t` without being truncated. Thus, the final total size value will not miss any overflow.

Reply:
I'm just wondering if this function is necessary to be in libcudf since it seems a simply enough check.

Reply:
~~I'm happy to keep it, but let remove it for brevity, assuming that I can reproduce it easily in the Spark plugin with minor overhead of casting a string sizes column from `int64_t` to `int32_t`.~~

There is a request from the python side for this API, and I'm not sure how they handle the overflow without the existence of this function, as well as whether they will ever have overflow with their datasets. @beckernick regarding the `repeat_strings` API in python, do you need to check for overflow in the output strings?

Reply:
I think that having this in libcudf is fine then.